src/llm/http_llm_calculator.cc

```diff
@@ -195,88 +193,81 @@ class OpenAIChatCompletionsRequest {
         it = this->doc.FindMember("repetition_penalty");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("repetition_penalty is not a double");
             this->repetitionPenalty = it->value.GetDouble();
-            // TODO: Validate?
         }
 
         // diversity_penalty: float; optional - defaults to 1.0
         // Extension, unsupported by OpenAI API and vLLM, however available in CB lib
         it = this->doc.FindMember("diversity_penalty");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("diversity_penalty is not a double");
             this->diversityPenalty = it->value.GetDouble();
-            // TODO: Validate?
         }
 
         // length_penalty: float; optional - defaults to 1.0
         // Extension, unsupported by OpenAI API however supported by vLLM and CB lib
         it = this->doc.FindMember("length_penalty");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("length_penalty is not a double");
             this->lengthPenalty = it->value.GetDouble();
-            // TODO: Validate?
         }
 
         // temperature: float; optional - defaults to 0.0 (different than OpenAI which is 1.0)
         it = this->doc.FindMember("temperature");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("temperature is not a double");
             this->temperature = it->value.GetDouble();
             if (this->temperature < 0.0f || this->temperature > 2.0f)
-                return false;
+                return absl::InvalidArgumentError("temperature out of range(0.0,2.0)");
         }
 
         // top_p: float; optional - defaults to 1
         it = this->doc.FindMember("top_p");
         if (it != this->doc.MemberEnd()) {
             if (!it->value.IsDouble())
-                return false;
+                return absl::InvalidArgumentError("top_p is not a double");
```

Comment:
I think we should go with a "floating point number" instead of a "double" since it's a more general term, better suiting for the API I believe.